content/local/writer.go

```diff
@@ -130,12 +134,13 @@ func (w *writer) Commit(size int64, expected digest.Digest, opts ...content.Opt)
 //
 // To abandon a transaction completely, first call close then `Store.Remove` to
 // clean up the associated resources.
-func (cw *writer) Close() (err error) {
-	unlock(cw.ref)
-
-	if cw.fp != nil {
-		cw.fp.Sync()
-		return cw.fp.Close()
+func (w *writer) Close() (err error) {
+	if w.fp != nil {
```

Comment:
I consider that the same, the same goroutine which is writing should be closing. I don't see patterns where the write and the write cleanup is being done on different non-synchronized goroutines, at least not ones we are concerned of supporting. Feel free to prove me wrong with a counter example 😄